.github/workflows/validate.yml

```diff
@@ -0,0 +1,91 @@
+name: Test and publish
+
+on:
+  pull_request:
+  push:
+    branches: [ main ]
+    tags: [ v.* ]
+
+jobs:
+  # share matrix between workflows, see https://github.community/t/how-to-share-matrix-between-jobs/128595/8
```

Comment:
Matrix sharing is probably not needed as publishing doesn't require it right now. Removing again...